packages/teams-js/test/public/thirdPartyCloudStorage.spec.ts

```diff
@@ -0,0 +1,207 @@
+    thirdPartyCloudStorage.getDragAndDropFiles('mockDragAndDropInput', mockCallback);
+    expect(sendMessageToParentSpy).toHaveBeenCalled();
+    const callbackused = sendMessageToParentSpy.mock.calls[0][2]; // calling the callback which was passed
+    if (callbackused) {
+      callbackused(mockFileResult);
+    }
+    expect(mockCallback).toHaveBeenCalledWith([], mockFileResult.error);
+  });
+
+  it('should call handleGetDragAndDropFilesCallbackRequest and the callback without error [single file]', async () => {
+    GlobalVars.isFramelessWindow = true;
+    await utils.initializeWithContext(FrameContexts.task, HostClientType.android);
+    utils.setRuntimeConfig({ apiVersion: 1, supports: { thirdPartyCloudStorage: {} } });
+
+    const sendMessageToParentSpy = jest.spyOn(communicationModule, 'sendMessageToParent');
+    thirdPartyCloudStorage.getDragAndDropFiles('mockDragAndDropInput', mockCallback);
+    expect(sendMessageToParentSpy).toHaveBeenCalled();
+    const callbackused = sendMessageToParentSpy.mock.calls[0][2]; // calling the callback which was passed
+    if (callbackused) {
+      // sending single file with 100 chunks
+      mockFileResults.forEach((mockFileResult) => {
+        callbackused(mockFileResult);
+      });
+      callbackused(mockFileResult2);
+    }
+
+    expect(mockCallback).toHaveBeenCalled();
+    expect(mockCallback).toHaveBeenCalledWith(expect.arrayContaining([expect.any(Blob)]), undefined); // verify we recieved 1 blob object i.e. one file
+  });
+
+  it('should call handleGetDragAndDropFilesCallbackRequest and the callback without error [multiple files]', async () => {
+    GlobalVars.isFramelessWindow = true;
+    await utils.initializeWithContext(FrameContexts.task, HostClientType.android);
+    utils.setRuntimeConfig({ apiVersion: 1, supports: { thirdPartyCloudStorage: {} } });
+
+    const sendMessageToParentSpy = jest.spyOn(communicationModule, 'sendMessageToParent');
+    thirdPartyCloudStorage.getDragAndDropFiles('mockDragAndDropInput', mockCallback);
+    expect(sendMessageToParentSpy).toHaveBeenCalled();
+    const callbackused = sendMessageToParentSpy.mock.calls[0][2]; // calling the callback which was passed
+    if (callbackused) {
+      // creating 50 file, each having 100 chunks
+      for (let i = 0; i < 50; i++) {
+        mockFileResults.forEach((mockFileResult) => {
+          callbackused(mockFileResult);
+        });
+        if (i == 49) {
+          callbackused(mockFileResult2);
+        } else {
+          callbackused(mockFileResult3);
+        }
+      }
+    }
+
+    expect(mockCallback).toHaveBeenCalled();
+    expect(mockCallback).toHaveBeenCalledWith(expect.arrayContaining(Array(50).fill(expect.any(Blob))), undefined);
+    const receivedArray = mockCallback.mock.calls[0][0];
+    expect(receivedArray).toHaveLength(50); // verify if we received 50 files
+  });
+
+  it('should call handleGetDragAndDropFilesCallbackRequest and the callback with error', async () => {
```

Comment:
Please add a unit test that calls `getDragAndDropFiles` twice and verify that the second call fails and throws an error

Reply:
Added test case for this scenario.